Yoast/javascript
# Ignore prefixes for Arabic keyphrase recognition

## Summary


This PR can be summarized in the following changelog entry:

* [yoastseo] Creates basic keyphrase forms for Arabic when they are preceded by one of the following prefixes:  `ف ل ب ال ك و`.

* [Yoast SEO Free] Makes it possible to recognize keyphrases in Arabic when they are preceded by the most common articles or prepositions (for example: `منزل` ("home") in `المنزل` ("the home")).

## Relevant technical choices:

* The current implementation is based on creating all possible prefixed forms, on the basis of [its Hebrew counterpart](https://github.com/Yoast/javascript/pull/780).

## Test instructions

This PR can be tested by following these steps:
* Run `yarn test` and make sure all tests pass.

* Test in the content-analysis app:
        * Start the content-analysis app by running `yarn start`
        * Paste a text of at least 150 words
        * Set the locale to `ar` 
        * Switch the morphology button to Off
        * Set the keyphrase as `منزل` and paste `المنزل` in the text 
        * Check in the assessment that the keyword is found in the text
        * Check that the functionality still works when the keyphrase and form in the text are switched around
        * Check that the keyphrase `منزل` matches also a form with a different prefix (e.g., `كمنزل`)



## QA Test instructions 
This PR can be tested by following these steps:
Test the following functionality in Free:

Set your site to Arabic 
As your keyphrase, set the word `منزل` ("home").
Add a text with at least 150 words.
In the text, add the form `المنزل` ("the home").
The keyphrase density assessment should show 1 keyphrase occurrence. When clicking on the marker for the assessment, `المنزل` should be marked.
Switch around the keyphrase and the word form.
The keyphrase density assessment should still show 1 keyphrase occurrence.
Switch the form in the text to a form with another prefix, e.g. `كمنزل` ("like a home").
The keyphrase density assessment should still show 1 keyphrase occurrence.
## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes

## Quality assurance


Fixes https://yoast.atlassian.net/browse/LIN-418
